packages/dashboard/src/components/dashboard/queryContext.ts

```diff
@@ -31,13 +39,28 @@ export const useQueries = ({
     assetModels,
     assets,
     properties,
+    alarms,
     requestSettings: {
       ...requestSettings,
       refreshRate,
     },
   });
 
-  const queries = [iotSiteWiseQuery.timeSeriesData(mappedQuery)] ?? [];
+  const queries = [];
+
+  // if the mapped query has an alarm query
+  if (mappedQuery.alarms?.length) {
+    queries.push(iotSiteWiseQuery.alarmData(mappedQuery));
+  }
+
+  // if the mapped query has timeSeries queries
+  if (
+    mappedQuery.assets.length ||
+    mappedQuery.assetModels?.length ||
+    mappedQuery.properties?.length
+  ) {
+    queries.push(iotSiteWiseQuery.timeSeriesData(mappedQuery));
+  }
```

Comment:
`assetModelQueryToSiteWiseAssetQuery` only changes the `assets` field of the query. Can we avoid using `mappedQuery.alarms`, `mappedQuery.assetModels`, and `mappedQuery.properties`? This would avoid you having to do the nullish coalescing since `alarms`, `assetModels`, and `properties` have already been defaulted as empty arrays. 

Similarly, `assetModelQueryToSiteWiseAssetQuery` should not take the whole query, where `query.assetModels` and `query.assets` may be undefined. It should take `assetModels` and `assets` as required and always return an array of asset queries, like:

```ts
export const assetModelQueryToSiteWiseAssetQuery({ 
  assetModels, 
  assets 
}: { 
  assetModels: WhateverAssetModelQueryTypeIs[]; 
  assets: WhateverAssetQueryTypeIs[]; 
}): WhateverAssetQueryTypeIs[] {}
```

If you do all this, then you don't need to use any if statements or nullish coalescing, and you can avoid pushing on the array, enabling more declarative code to be written, like:

```ts
const queries = [
  ...iotSiteWiseQuery.alarmData({ .. }),
  ...iotSiteWiseQuery.timeSeriesData({ .. }),
]
```

Reply:
I don't know if this will be possible since `iotSiteWiseQuery.alarmData` always returns an object. If the query passed to that function is undefined, the return value will be an object with `query: undefined`. But what we could do is something like this:

```
const queries = [
  ...[...assets, ...properties].map(query => iotSiteWiseQuery.timeSeriesData(query)),
  ...[...alarms].map(query => iotSiteWiseQuery.alarmData(query)),
]
```

Reply:
But yes, I think those assetModel / alarmAssetModel transformers could just take assetModel + assets / alarmAssetModel + alarms, and out only assets / alarms. Since we don't use assetModel / alarmAssetModel after those functions.